system/default_ad_api_helpers/ad_api_adaptors/config/initial_pose.param.yaml

```
@@ -0,0 +1,13 @@
+/**:
+  ros__parameters:
+
+    # from initialpose (Rviz's 2DPoseEstimate)
+    initial_pose_particle_covariance:
+      [
+        4.0, 0.0, 0.0,  0.0,  0.0,  0.0,
+        0.0, 4.0, 0.0,  0.0,  0.0,  0.0,
+        0.0, 0.0, 0.01, 0.0,  0.0,  0.0,
```

[nits] Since it's 2D, the Z covariance would be extremely large. I'll leave it to you whether to apply this or not.

```suggestion
        0.0, 0.0, 1000000, 0.0,  0.0,  0.0,
```

---

I think that if the variance is too large it will not work as an initial position. In the current implementation, most of the particles will not converge to the correct position, as they will be scattered over a wide area up and down. So a lot of particles and computation time are required. We can trust Z because it is fitted to the map.

---

Hmm, is this the covariance after the map fitting? Then it's okay.